tests/test_datasets/test_datasets.py

```diff
@@ -874,3 +874,191 @@ def test_extra_repr(self):
+            'tasks': [{
+                'name': 'gender',
+                'categories': ['a', 'b'],
+            }]
+        }
+        with self.assertRaisesRegex(AssertionError, 'Missing "type"'):
+            dataset_class(**cfg)
+
+        tmpdir.cleanup()
+
+    def test_data_root(self):
+        dataset_class = DATASETS.get(self.DATASET_TYPE)
+
+        # Test relative ann_file
+        cfg = {**self.DEFAULT_ARGS, 'ann_file': 'multi-task.json'}
+        dataset = dataset_class(**cfg)
+        self.assertEqual(dataset.data_root, ASSETS_ROOT)
+        self.assertEqual(dataset.ann_file,
+                         osp.join(dataset.data_root, 'multi-task.json'))
+
+        # Test relative data_prefix
+        cfg = {**self.DEFAULT_ARGS, 'data_prefix': 'train'}
+        dataset = dataset_class(**cfg)
+        self.assertEqual(dataset.data_prefix,
+                         osp.join(dataset.data_root, 'train'))
+
+        # Test no data_prefix
+        cfg = {**self.DEFAULT_ARGS, 'data_prefix': None}
+        dataset = dataset_class(**cfg)
+        self.assertEqual(dataset.data_prefix, dataset.data_root)
+
+    def test_parse_data_info(self):
+        dataset_class = DATASETS.get(self.DATASET_TYPE)
+        dataset = dataset_class(**self.DEFAULT_ARGS)
+
+        data = dataset.parse_data_info({
+            'img_path': 'a.jpg',
+            "gt_label":{"gender": 0}
+        })
+        self.assertDictContainsSubset(
+            {
+                'img_path': ASSETS_ROOT+'/a.jpg',
+                'gender': 0
+            }, data)
+        np.testing.assert_equal(data['gender'],0)
+
+
+        # Test invalid type
+        with self.assertRaisesRegex(AssertionError, "got <class 'str'>"):
+            dataset.parse_data_info('hi')
+
+        # Test missing path
+        with self.assertRaisesRegex(AssertionError, 'have `img_path` field'):
+            dataset.parse_data_info({
+                "gt_label":{
+                    'gender': 0,
+                    'wear': [1, 0, 1, 0]
+                    }
+            })
+
+
+        # Test invalid label type
+        with self.assertRaisesRegex(AssertionError, 'a single number'):
+            dataset.parse_data_info({
+                'img_path': 'a.jpg',
+                "gt_label":{
+                    'gender': [0, 1],
+                    'wear': [1, 0, 1, 0]
+                    }
+            })
+        with self.assertRaisesRegex(AssertionError, 'one-hot format'):
+            dataset.parse_data_info({
+                'img_path': 'a.jpg',
+                "gt_label":{
+                    'gender': 0,
+                    'wear': [1, 2, 3, 4]
+                }
+            })
+
+    def test_get_cat_ids(self):
+        dataset_class = DATASETS.get(self.DATASET_TYPE)
+        dataset = dataset_class(**self.DEFAULT_ARGS)
+
+        cat_ids = dataset.get_cat_ids(1)
+        self.assertIsInstance(cat_ids, dict)
+        self.assertDictEqual(cat_ids, dict(gender=[0], wear=[0, 2]))
+
```

Comment:
Add a method `test_load_data_list`

Reply:
test the perse_info , it's better because load_data_list is just a loop that uses parse

Reply:
1. it is called test_load_data_list for other datasets, so i feel it's better to follow the same convention
2. I'm not sure this is the same, in parse_info you are testinig `dataset.parse_data_info`, i'm expecting the unit test to test `len(dataset)`, `dataset[0]` in different situation as it is made for CIFAR10, for example in https://github.com/marouaneamz/mmclassification/blob/issue7/tests/test_datasets/test_datasets.py#L384-L391